src/cpp/continuous_batching/src/sampler.hpp

```diff
@@ -110,14 +110,17 @@ struct Group {
     std::vector<Beam> min_heap;  // The worst of the best completed beams is the first
     bool done = false;
 
-    int64_t finish(Beam beam, const GenerationConfig& sampling_params) {
+    int64_t finish(Beam beam, const ov::genai::GenerationConfig& sampling_params) {
         int64_t preeempted_sequence_id = -1;
         float generated_len = beam.get_generated_len() + (beam.m_token_id == sampling_params.eos_token_id ? 1 : 0); // HF counts EOS token in generation length
         beam.m_score /= std::pow(generated_len, sampling_params.length_penalty);
 
         min_heap.push_back(beam);
         std::push_heap(min_heap.begin(), min_heap.end(), greater);
-        if (min_heap.size() > sampling_params.group_size) {
+        OPENVINO_ASSERT(sampling_params.num_beams % sampling_params.num_beam_groups == 0,
+            "number of beams should be divisible by number of groups");
```

Comment:
such checks must be in validation of generation config. If they present there, let's drop from current places